### DISCUSSION TITLE
Home Asistant and TasmoAdmin

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#18279] [Q&A] [answered]
Hello, I recently setup Home Assistant on Raspberry pie and installed TasmoAdmin add-on. The add-on seems to be working nicely however I noticed that the devices stop responding on TasmoAdmin when I set any one of the devices to "webserver 0 or 1". Devices are only active on TasmoAdmin when "webserver 2" is set. I suspect TasmoAdmin is generating lot of extra traffic on the network trying to poll 5 devices through webserver. Is there any other program that gives similar functionality but uses MQTT? and is it advisable to leave webserver on 2 just for TasmoAdmin?

**Comment by sfromis** [comment, marked as answer]
Well, TasmoAdmin is not part of the Tasmota project, and not a tool we even like to recommend, in part due to it generating a lot of extra traffic when polling. The tool [TDM](https://tasmota.github.io/docs/Tasmota-Device-Manager/) has a similar scope, without polling all the time.

What is "advisable" is up to what particular "philosophy" you have about such, but on my home network, I dont consider micromanagement like turning off webserver useful. Of course, it is up to you how you like to have nerdy fun :wink:

**Reply by NsinghP** [reply]
Thanks, I will try out TDM.

**Reply by NsinghP** [reply]
> ... The tool [TDM](https://tasmota.github.io/docs/Tasmota-Device-Manager/) has a similar scope, without polling all the time.

Dowloaded TDM for windows and realized that it will not  work with HiveMQ Cloud. I will have to setup a local broker (That may not be hard to do since I have Raspberry Pi running Home Assistant) unless there is a version of TDM with MQTT-TLS.

**Reply by NsinghP** [reply]
Setting up local MQTT broker add-on on Home Assistant was a breeze. Now I have "TDM" working like a charm on a local broker and I don't even need MQTT-TLS. So all the time I spent updating my custom tasmota devices to use MQTT-TLS was an exercise in futility :( but no harm done, I learned a thing or two about how MQTT works.

**Reply by sfromis** [reply]
Yeah, I was wondering why you were running a cloud broker when having a machine with a HA setup....

Anyway, I'd say that "best practice" for a local MQTT broker is to **not** run it as a HA add-on, but instead have it running separately. The point is that you may have many cases where you need to restart HA, and then it creates much less disturbance if you keep the broker running 100% instead of such outages.

**Reply by NsinghP** [reply]
>... What is "advisable" is up to what particular "philosophy" you have about such, but on my home network, I dont consider micromanagement like turning off webserver useful. Of course, it is up to you how you like to have nerdy fun 😉

The reason I am so touchy about securing my devices is simple-I live in a multi storied building with 4 Apartments per floor and about 3 more such multi storied buildings with more apartments in close proximity. On any given day I have multiple Wifi networks that are visible and occasionally some that are open! Since Tasmota webserver does not use SSL, I just feel nervous leaving it on.  Also I have a major concern about AP function that kicks in when any device gets reset.

**Reply by sfromis** [reply]
The latter point about a device getting reset is moot, as that would also reset the WebServer selection. Do not use `WifiConfig 2`, as this opens up for simple deauth attacks. You are obviously *craving* trouble if you *ever* have an open AP on your network.

You could also do your own builds with a preconfigured Wifi+MQTT setup, and web server disabled, and review what recovery features of Tasmota you want - or not.

**Reply by NsinghP** [reply]
> You could also do your own builds with a preconfigured Wifi+MQTT setup, and web server disabled, and review what recovery features of Tasmota you want - or not.


That is exactly what I have done with following Custom Parameters: Feel free to make more recommendations or point out any errors:

#define MODULE            USER_MODULE
#define USER_TEMPLATE "{\"NAME\":\"Redacted Tasmota\",\"GPIO\":[259,256,32,160,258,257,0,0,162,163,161,0,0,0],\"FLAG\":0,\"BASE\":18}"
#define EMULATION         EMUL_HUE
#define USE_EMULATION_HUE
#define DEVICE_NAME       "First Room"  
#define FRIENDLY_NAME     "First Room Switch"
#define SERIAL_LOG_LEVEL LOG_LEVEL_NONE
#define WIFI_AP_PASSPHRASE "Redacted Tasmota Rocks" 
#define PROJECT            "Redacted Tasmota"
#define USE_TASMOTA_DISCOVERY
#define OTA_URL "http://please-consult-Network-Admin-/" PROJECT "-"  CODE_IMAGE_STR ".bin.gz"
#define WEB_PASSWORD       "ReDaCtEd!-9-1-8-2"

**Reply by barbudor** [reply]
May be related to the copy paste but your USER_TEMPLATE is missing `\` in front of each `"` internal to the string

**Reply by NsinghP** [reply]
> May be related to the copy paste but your USER_TEMPLATE is missing `\` in front of each `"` internal to the string

Thanks for pointing that out, and yes it is copy and paste related. I remember going through this problem initially when I first started posting in this forum couple of years ago. (By The way most of the custom strings above were generated with your help, Thanks!)
